apps/meteor/client/views/admin/moderation/hooks/useDeleteMessage.tsx

```diff
@@ -4,20 +4,36 @@ import React from 'react';
 
 import GenericModal from '../../../../components/GenericModal';
 
+function isApiResponse(response: unknown): response is { success: boolean } {
+	return typeof response === 'object' && response !== null && 'success' in response;
+}
+
 const useDeleteMessage = (mid: string, rid: string, onChange: () => void) => {
 	const t = useTranslation();
+	const getMessage = useEndpoint('GET', '/v1/chat.getMessage');
 	const deleteMessage = useEndpoint('POST', '/v1/chat.delete');
 	const dismissMessage = useEndpoint('POST', '/v1/moderation.dismissReports');
 	const dispatchToastMessage = useToastMessageDispatch();
 	const setModal = useSetModal();
 	const queryClient = useQueryClient();
 
 	const handleDeleteMessages = useMutation({
-		mutationFn: deleteMessage,
+		mutationFn: async ({ msgId, roomId, asUser }: { msgId: string; roomId: string; asUser: boolean }) => {
+			try {
+				await getMessage({ msgId });
+			} catch (error) {
+				return error;
+			}
+			return deleteMessage({ msgId, roomId, asUser });
```

Comment:
is this really the way?
I'm afraid sometimes the user will not be able to get the message through `chat.getMessage` or even delete the message using 'chat.detele' for permission reasons but still able to moderate, maybe the we should use the same logic from `/v1/moderation.user.deleteReportedMessages` but only for one message

the hook `useDeleteMessagesAction` looks ideal btw

Reply:
We basically want to handle a specific not-found situation for chat.delete. For any other case, block the dismiss action. This does make the assumption that user x can read all message. 

> the hook useDeleteMessagesAction looks ideal btw

that I think is too much? delete all message for a user vs a single random message. 

At this point i'm thinking modifying the dismissReport endpoint to pass an optional like `deleteMessages` to first delete (or ignore if doesn't exist) then mark hidden

Reply:
Yes, that had be good. But, I would prefer to not modify the API endpoints now, considering the release deadline 😬.

Reply:
Don't think you need to worry about that too much.

Reply:
It makes sense, and there might be cases where the moderator might not have access to messages, but as it is now, the Moderation Console is not yet available for scoped roles, so it might be very few cases where it does happen. 

And the main concern for opening this fix was that if there was any such error while deleting a single message, the error was shown as a toast, but the confirmation modal was not getting closed. And despite the message not being deleted, the reports against that message were cleared out, which is not something good, as per my understanding.

And after this fix, the delete confirmation model will get closed in case of any error on trying to delete the message. But the reports won't be dismissed; in the meanwhile, the user who was trying to perform this moderation action can figure out why there was an error, resolve it, come back, and delete it again. As a last resort, if it is some strange unknown issue, they can force dismiss the reported message using the 'Dismiss Reports' action. So, as far as I could see, it is all covered. I am open to suggestions, in case, I overlooked any cases.

Thank you!